src/main/java/org/broadinstitute/hellbender/tools/sv/SVClusterEngine.java

```diff
@@ -0,0 +1,205 @@
+        final List<String> algorithms = cluster.stream().flatMap(v -> v.getAlgorithms().stream()).distinct().collect(Collectors.toList());
+        final List<Genotype> clusterSamples = cluster.stream().flatMap(v -> v.getGenotypes().stream()).collect(Collectors.toList());
+
+        final int newStart;
+        final int newEnd;
+        if (exampleCall.getType().equals(StructuralVariantType.INS)) {
+            // Insertions should be a single locus; also fixes case where end-supporting split reads are to the
+            // left of start-supporting split reads
+            final int mean = (medianStart + medianEnd) / 2;
+            newStart = mean;
+            newEnd = mean + 1;
+        } else {
+            newStart = medianStart;
+            newEnd = medianEnd;
+        }
+        //??? does evidence not need to be merged???
+        return new SVCallRecordWithEvidence(exampleCall.getContig(), newStart, exampleCall.getStartStrand(),
+                exampleCall.getEndContig(), newEnd, exampleCall.getEndStrand(), exampleCall.getType(), length, algorithms, clusterSamples,
+                exampleCall.getStartSplitReadSites(), exampleCall.getEndSplitReadSites(), exampleCall.getDiscordantPairs());
+    }
+
+    @Override
+    protected boolean clusterTogether(final SVCallRecordWithEvidence a, final SVCallRecordWithEvidence b) {
+        if (!a.getType().equals(b.getType())) return false;
+        final boolean depthOnlyA = isDepthOnlyCall(a);
+        final boolean depthOnlyB = isDepthOnlyCall(b);
+        if (depthOnlyA && depthOnlyB) {
+            return clusterTogetherBothDepthOnly(a, b);
+        } else if (depthOnlyA != depthOnlyB) {
+            return clusterTogetherMixedEvidence(a, b);
+        } else {
+            return clusterTogetherBothWithEvidence(a, b);
+        }
+    }
+
+    /**
+     * Determine an overlap interval for clustering using reciprocal overlap or breakend window, as applicable
+     * Returned interval represents the interval in which the start position of a new event must fall in order to be added to the cluster (including {@param call})
+     * @param call  new event to be clustered
+     * @param clusterMinStartInterval    the cluster of interest, may be null
+     * @return  an interval describing the cluster after {@param call} is added
+     */
+    @Override
+    protected SimpleInterval getClusteringInterval(final SVCallRecordWithEvidence call, final SimpleInterval clusterMinStartInterval) {
+        final int minStart;
+        final int maxStart;
+        if (isDepthOnlyCall(call)) {
+            minStart = (int) (call.getEnd() - call.getLength() / MIN_RECIPROCAL_OVERLAP_DEPTH); //start of an overlapping event such that call represents (reciprocal overlap) of that event
+            maxStart = (int) (call.getStart() + (1.0 - MIN_RECIPROCAL_OVERLAP_DEPTH) * call.getLength());
```

Comment:
I modified the math here to better reflect my idea of how we compute reciprocal overlap, but I'm open to discussion.